### PR TITLE
build: fixed GitHub actions on usernames with uppercase characters

### PR DESCRIPTION
**Pull request by JohnTheNerd**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

Fixed GitHub actions for everyone that has uppercase characters in their username, by converting the image name to lowercase. GHCR image names are required to be in lowercase.

### Added

N/A

### Changed

N/A

### Deprecated

N/A

### Removed

N/A

### Fixed

GitHub Actions is now fixed for anyone with uppercase characters in their username

### Security

N/A

### Breaking Changes

N/A

---

### Additional Information

You can see the actions running in my fork. Discussion post is #3203 

### Screenshots or Videos

image

### CODE REVIEW

File: .github/workflows/docker-build.yaml [hunk 1]
```diff
@@ -28,6 +26,15 @@ jobs:
...
 
     steps:
+      # GitHub Packages requires the entire repository name to be in lowercase
+      # although the repository owner has a lowercase username, this prevents some people from running actions after forking
+      - name: Set repository and image name to lowercase
+        run: |
+          echo "IMAGE_NAME=${IMAGE_NAME_MIXED_CASE,,}" >>${GITHUB_ENV}
+          echo "FULL_IMAGE_NAME=ghcr.io/${IMAGE_NAME_MIXED_CASE,,}" >>${GITHUB_ENV}
+        env:
+          IMAGE_NAME_MIXED_CASE: '${{ github.repository }}'
```
Comment (tjbck) [inline .github/workflows/docker-build.yaml:36]: Let's just reuse the same env var name here.

`IMAGE_NAME: '${{ github.repository }}'`
`echo "IMAGE_NAME=${IMAGE_NAME,,}" >>${GITHUB_ENV}`

The rest LGTM!

Comment (JohnTheNerd) [reply .github/workflows/docker-build.yaml]: done - [just testing it now](https://github.com/JohnTheNerd/open-webui/actions/runs/9537871843)

Comment (JohnTheNerd) [reply .github/workflows/docker-build.yaml]: the build succeeded - resolving the conversation now

**Comment by tjbck** [comment]
LGTM, Thanks!

**Pull request merged**